### ISSUE TITLE
when searching through a type, that it shortlists the type's via values in `Description` and `PredefinedType`, or `ElementType`, in addition to `Type.Name`. 

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#5166]
Would be nice, when searching through a type, that it shortlists the type's via values in `Description` and `PredefinedType`, or `ElementType`, in addition to `Type.Name`. 

https://github.com/user-attachments/assets/e820ecc3-dc94-44be-91fc-303eabae580f

Similar to how when searching through IfcClasses, the search results return the PredefinedType as well.

https://github.com/user-attachments/assets/e0f23b17-aa50-4ea3-963e-e5232c6cc88a

**Comment by theoryshaw** [comment]
:)

**Comment by theoryshaw** [comment]
@falken10vdl Was something like this implemented elsewhere?  I remember there was some similar work in this area?

**Comment by falken10vdl** [comment]
Yes. I think that in a way it would be possible similar to https://github.com/IfcOpenShell/IfcOpenShell/issues/6822
Let me take a look and come back to you.
Cheers!

**Comment by theoryshaw** [comment]
That would be awesome! Since the 'Name' attribute is [usually short](https://community.osarch.org/discussion/1627/what-pset-or-attribute-do-you-think-would-be-the-equivalent-of-revits-type-comments) (a number or code), it's hard sometimes finding what you're looking for.

**Comment by falken10vdl** [comment]
@theoryshaw Please take a look to https://github.com/IfcOpenShell/IfcOpenShell/pull/6962
Is this what you are looking for? I had not a very comprehensive ifc file with a good bunch of examples to test. :)
I also have joined the suggestions in single line Name > Description > PredefinedType > ElementType > ObjectType instead of having many lines like Name > Description, Name > PredefinedType, etc.

**Comment by theoryshaw** [comment]
Oh man, this works awesome! Thank you.

...

Would suggest having the same functionality here as well.

<img width="1588" height="665" alt="Image" src="https://github.com/user-attachments/assets/04ceb067-a1c3-4756-bd9b-9115a3b145e6" />

**Comment by falken10vdl** [comment]
@theoryshaw updated the PR to include relating_type_id which is the one used in that property.
Cheers!

**Comment by falken10vdl** [comment]
@theoryshaw Are we happy to close this one?
Cheers!

**Comment by theoryshaw** [comment]
Yes, thanks so much!
